lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -294,10 +351,9 @@ static int trace_deinit(void)
 {
 	int err;
 
-	err = trace_backend_deinit();
+	err = trace_backend.deinit();
```

Comment:
I would add a check to see if the interface function is implemented for every function, even the "default" ones like init, deinit. Backends are potentially implemented by third-parties and they should get an immediate error if they forgot to implement the deinit.

Reply:
Unless it's enforced at compiled time that `init`, `deinit` and `write` are mandatory (but I don't see any compile-time enforcement of this).

Reply:
I'll add a check for this in `trace_init`.